### PR TITLE
quiet compiler, remove some trailing control-Ms

### PR DESCRIPTION
**Pull request by sethalves**

### CODE REVIEW

**Comment by sethalves** [comment]
I mainly want to make sure that *everyone* has a rough merge on Monday.

**Comment by hifi-gustavo** [comment]
Build console output for commit: 0fb1a83e3bc4956d12e09c48994430c9b4ee864f available at: https://gist.github.com/b0efdad9711d7c9b6bcb<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4697.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4697.dmg

**Comment by Atlante45** [comment]
Ahah, I did the same but you kinda took it to the next level ^^

File: interface/src/Application.cpp [hunk 1]
```diff
@@ -1058,7 +1058,7 @@ void Application::keyPressEvent(QKeyEvent* event) {
         bool isShifted = event->modifiers().testFlag(Qt::ShiftModifier);
         bool isMeta = event->modifiers().testFlag(Qt::ControlModifier);
         bool isOption = event->modifiers().testFlag(Qt::AltModifier);
-        bool isKeypad = event->modifiers().testFlag(Qt::KeypadModifier);
+        // bool isKeypad = event->modifiers().testFlag(Qt::KeypadModifier);
```
Comment (Atlante45) [inline interface/src/Application.cpp]: Coding standard: remover line

File: interface/src/ui/ApplicationOverlay.cpp [hunk 2]
```diff
@@ -257,7 +257,7 @@ void ApplicationOverlay::displayOverlayTexture() {
     if (_alpha == 0.0f) {
         return;
     }
-    auto glCanvas = Application::getInstance()->getGLWidget();
+    // auto glCanvas = Application::getInstance()->getGLWidget();
```
Comment (Atlante45) [inline interface/src/ui/ApplicationOverlay.cpp]: Coding standard: remover line

File: libraries/gpu/src/gpu/GLBackendOutput.cpp [hunk 3]
```diff
@@ -24,7 +24,7 @@ GLBackend::GLFramebuffer* GLBackend::syncGPUObject(const Framebuffer& framebuffe
     GLFramebuffer* object = Backend::getGPUObject<GLBackend::GLFramebuffer>(framebuffer);
 
     // If GPU object already created and in sync
-    bool needUpdate = false;
+    // bool needUpdate = false;
```
Comment (Atlante45) [inline libraries/gpu/src/gpu/GLBackendOutput.cpp]: Coding standard: remover line

File: libraries/gpu/src/gpu/GLBackendTexture.cpp [hunk 4]
```diff
@@ -176,6 +176,10 @@ class GLTexelFormat {
                         texel.internalFormat = GL_DEPTH_COMPONENT24;
                         break;
                         }
+                    case gpu::NUM_TYPES: { // quiet compiler
+                        assert(false);
+                        break;
```
Comment (Atlante45) [inline libraries/gpu/src/gpu/GLBackendTexture.cpp]: There's a Qt macro `Q_UNREACHABLE();` that's made just for that.

File: libraries/render-utils/src/Model.cpp [hunk 5]
```diff
@@ -111,7 +111,8 @@ void Model::RenderPipelineLib::addRenderPipeline(Model::RenderKey key,
     slotBindings.insert(gpu::Shader::Binding(std::string("emissiveMap"), 3));
 
     gpu::ShaderPointer program = gpu::ShaderPointer(gpu::Shader::createProgram(vertexShader, pixelShader));
-    bool makeResult = gpu::Shader::makeProgram(*program, slotBindings);
+    // bool makeResult =
```
Comment (Atlante45) [inline libraries/render-utils/src/Model.cpp]: Coding standard: remover line

**Comment by Atlante45** [comment]
How to look stupid:
Step 1: Make a comment on a PR with an error in it.
Step 2: Copy paste that comment wherever appropriate.
Step 3: Notice error and proceed to go and edit each comments.

File: libraries/gpu/src/gpu/GLBackend.cpp [hunk 6]
```diff
@@ -143,7 +143,7 @@ void GLBackend::do_draw(Batch& batch, uint32 paramOffset) {
     uint32 startVertex = batch._params[paramOffset + 0]._uint;
 
     glDrawArrays(mode, startVertex, numVertices);
-    CHECK_GL_ERROR();
+    (void) CHECK_GL_ERROR();
```
Comment (jherico) [inline libraries/gpu/src/gpu/GLBackend.cpp]: I suspect there's an easier way to do this, by modifying the #define for CHECK_GL_ERROR to be something like
```
#ifdef DEBUG
#define CHECK_GL_ERROR ...
#else
#define CHECK_GL_ERROR (void)
#endif

Comment (sethalves) [inline libraries/gpu/src/gpu/GLBackend.cpp]: There are places (or, at least there were places) where it's used in a if statement.

**Comment by hifi-gustavo** [comment]
Build console output for commit: b69cba1aba174da4b1f368db54cf632e685cb80b available at: https://gist.github.com/b0efdad9711d7c9b6bcb<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4697.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4697.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 2eebfb9cf1983ed9be99c01869ae2a11f3062e38 available at: https://gist.github.com/b0efdad9711d7c9b6bcb<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4697.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4697.dmg

File: libraries/networking/src/DataServerAccountInfo.cpp [hunk 7]
```diff
@@ -17,7 +17,9 @@
 #include "NetworkLogging.h"
 #include "DataServerAccountInfo.h"
 
+#ifndef __GNUC__
```
Comment (jherico) [inline libraries/networking/src/DataServerAccountInfo.cpp]: is this supposed to be `#ifdef`?

Comment (sethalves) [inline libraries/networking/src/DataServerAccountInfo.cpp]: Heh well...
I did intend ifndef.   I build with g++ and it complains about that pragma.  If it doesn't work for msvc, then who is it for?  Should we just remove it?

Comment (jherico) [inline libraries/networking/src/DataServerAccountInfo.cpp]: I added the pragma because without it OSX goes nuts with deprecation warnings about SSL.

Comment (ZappoMan) [inline libraries/networking/src/DataServerAccountInfo.cpp]: Sounds like you need to upgrade your SSL libraries. That's the correct fix to silencing those warnings.

**Pull request merged**